File: sefetylock-client/src/main/java/org/entities/Message.java
package org.entities;

import com.google.gson.Gson;
import java.io.Serializable;
import java.util.UUID;

public class Message implements Serializable {
    private String id;
    private String type;
    private String method;
    private String objectReference;
    private Object arguments;

    public Message() {
        this.id = generateMessageId();
    }

    // Gera um UUID aleatório para userId
    private String generateMessageId() {
        return UUID.randomUUID().toString();
    }

    // Getters e setters
    public String getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getMethod() {
        return method;
    }

    public void setMethod(String method) {
        this.method = method;
    }

    public String getObjectReference() {
        return objectReference;
    }

    public void setObjReference(String objectReference) {
        this.objectReference = objectReference;
    }

    public Object getArguments() {
        return arguments;
    }

    public void setArguments(Object arguments) {
        this.arguments = arguments;
    }

    public String toJson() {
        return new Gson().toJson(this);
    }
}

File: safetylock-server/entities/message.py
class Message:
    def __init__(self, id=None, type=None, method=None, objectReference=None, arguments=None):
        self.id = id
        self.type = type
        self.method = method
        self.objectReference = objectReference
        self.arguments = arguments

    @classmethod
    def from_dict(cls, message_dict):
        return cls(
            message_dict.get('id'),
            message_dict.get('type'),
            message_dict.get('method'),
            message_dict.get('objectReference'),
            message_dict.get('arguments')
        )

    def to_dict(self):
        return {
            'id': self.id,
            'type': self.type,
            'method': self.method,
            'objectReference': self.objectReference,
            'arguments': self.arguments
        }
